app/src/main/java/com/spendster/presentation/validation/ComplexPhoneValidator.java

```diff
@@ -0,0 +1,28 @@
+package com.spendster.presentation.validation;
+
+public class ComplexPhoneValidator implements Validation<ValidationResource>  {
+    private String phone;
+
+    public ComplexPhoneValidator(String phone) {
+        this.phone = phone;
+    }
+
+    @Override
+    public ValidationResource validate() {
+        boolean isValid = true;
+        String errorMessage = "";
+        if(this.phone.isEmpty()){
+            errorMessage = "First and Last name field is empty";
+            isValid = false;
+        }
+        else if (this.phone.length() < 10){
```

Comment:
@SimonMaha this is not a perfect validation of phone number as in different countries the length might be different. Please investigate phone number validation and sync with IOS and server on this.